the-collab-lab/tcl-45-smart-shopping-list
# Nz mobile view list

## Description

Hi everyone,

This is a huge PR, because I put all ideas I had in mind into it.



1. This PR intends to make one page of the the app mobile friendly by making bigger font, buttons, changing the layout to avoid any overlapping. I am talking about the List view.
2. I also changed the way the items are distributed between time frames by adding a simpler filter intend of the config. In this PR only time frames which have items in them are displayed. In the previous version, even empty time frames (e.g. 'Inactive') were displayed.
3. When the item is purchased its label text becomes strikethrough 
4. The time frames in the list view were changed back to more vague soon-kind of soon-not so soon

## Related Issue


Closes #74 

## Type of Changes



|     | Type                       |
| --- | -------------------------- |
| ✓    | :bug: Bug fix              |
| ✓   | :sparkles: New feature     |
|  ✓   | :hammer: Refactoring       |
|     | :100: Add tests            |
|     | :link: Update dependencies |
|     | :scroll: Docs              |

## Updates

### Before

Empty time frame displayed:

<img width="385" alt="Screen Shot 2022-09-28 at 12 42 20 PM" src="https://user-images.githubusercontent.com/63440229/192864053-74c7c39b-efb2-41b3-9f1f-70ab3c02ed05.png">

Item content stretched in the middle, buttons in the header not fully displayed:

<img width="386" alt="Screen Shot 2022-09-28 at 12 42 08 PM" src="https://user-images.githubusercontent.com/63440229/192864059-16a5a04c-e635-4990-a94a-8de6e16adc33.png">


### After

<img width="385" alt="Screen Shot 2022-09-28 at 1 01 42 PM" src="https://user-images.githubusercontent.com/63440229/192866833-3e5acd43-7ae9-4b09-a1ce-92178ec8769a.png">



### What can be improved:

The content of each item box is centred leading to the ghost image and the button block with Delete and Edit move right and left depending on the label length. I am currently looking for a way to make the position of the button block and the image fixed along invisible lines to make the app look more appealing.

## Testing Steps / QA Criteria

1. Check out the branch `NZ-mobile-view` and run `npm start`
2. Add items to your list and go to the list view
3. Open DevTools in your browser and make the screen with the app smaller. Change dimensions to match smaller devices from iPhones to tables
